qualtran/bloqs/bookkeeping/free.py

```diff
@@ -69,10 +73,10 @@ def decompose_bloq(self) -> 'CompositeBloq':
     def adjoint(self) -> 'Bloq':
         from qualtran.bloqs.bookkeeping.allocate import Allocate
 
-        return Allocate(self.dtype)
+        return Allocate(self.dtype, self.dirty)
 
     def on_classical_vals(self, reg: int) -> Dict[str, 'ClassicalValT']:
-        if reg != 0:
+        if reg != 0 and not self.dirty:
             raise ValueError(f"Tried to free a non-zero register: {reg}.")
```

Comment:
Perhaps add the state of dirty to the error message.

Reply:
done